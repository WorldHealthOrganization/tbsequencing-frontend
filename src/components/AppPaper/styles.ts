import { css } from '@emotion/react';
import appColors from '../../styles/colors';

export const container = css({
  padding: 24,
  boxShadow: '0px 0px 4px rgba(201, 221, 234, 0.5)',
  backgroundColor: appColors.neutral.white,
});
