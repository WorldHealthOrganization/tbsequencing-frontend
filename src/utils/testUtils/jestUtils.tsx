import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
