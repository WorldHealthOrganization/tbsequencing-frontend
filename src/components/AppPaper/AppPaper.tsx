/** @jsxImportSource @emotion/react */
import React, { ReactNode } from 'react';
import { SerializedStyles } from '@emotion/react';
import * as styles from './styles';

interface Props {
  children: ReactNode;
  style?: SerializedStyles;
}

export const AppPaper = ({ children, style }: Props) => (
  <div css={[styles.container, style]}>
    {children}
  </div>
);

export default AppPaper;
