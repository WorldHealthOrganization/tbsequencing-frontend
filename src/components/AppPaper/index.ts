import AppPaper from './AppPaper';

export default AppPaper;
