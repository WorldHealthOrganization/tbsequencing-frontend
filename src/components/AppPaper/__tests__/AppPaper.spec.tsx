import React from 'react';
import { getTreeJSONWithRedux, render } from '../../../utils/testUtils/jestUtils';
import AppPaper from '../AppPaper';

describe('AppPaper component test', () => {
  test('renders AppPaper component without error', () => {
    render(
      <AppPaper>Mock</AppPaper>,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <AppPaper>Mock</AppPaper>,
    );

    expect(tree).toMatchSnapshot();
  });
});
