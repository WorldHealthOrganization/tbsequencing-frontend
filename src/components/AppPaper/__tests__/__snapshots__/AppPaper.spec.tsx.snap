// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPaper component test Should render correctly 1`] = `
<div
  className="css-1y50k9a-AppPaper"
>
  Mock
</div>
`;
